### PR TITLE
scripts: add support for setting `ICE_CPPDEFS` in set_env.* options

### PR DESCRIPTION
**Pull request by phil-blain**
## PR checklist
- [x] Short (1 sentence) summary of your PR: 
    add support for setting `ICE_CPPDEFS` in set_env.* options
- [x] Developer(s): 
    P. Blain
- [x] Suggest PR reviewers from list in the column to the right.
- [x] Please copy the PR test results link or provide a summary of testing completed below.
    I tested that the feature works by adding a `set_env.lapack` file with content `setenv ICE_CPPDEFS CICE_USE_LAPACK`
    and checking that the CPP is indeed used at compile time.
- How much do the PR code changes differ from the unmodified code? 
    - [x] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [x] No 
- [x] Please provide any additional information or relevant details below:

Allow using the present case/test configuration mechanism
(set_env.* in configuration/scripts/options/) to add preprocessor macros.

cice.setup already allows for modifying cice.setttings using set_env.*
options, so the only change needed is to add a default value for
`ICE_CPPDEFS` in the default cice.settings file, and not resetting
this variable in cice.build.

Note that contrary to what was proposed in CICE-Consortium/CICE#445, we
do not implement a similar ICE_SLIBS variable, that could be used to add
linker flags (such as optional libraries), because the way in which
libraries are made accessible and linked with an executable vary
depending on the machine in use, so such modifications are best made in
the `Macros.<machine>_<compiler>` and `env.<machine>_<compiler>` files.

### CODE REVIEW

File: configuration/scripts/cice.batch.csh [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #! /bin/csh -f
 
 if ( $1 != "" ) then
-  echo ${0:t} ${1}
+  echo "running cice.batch.csh (creating ${1})"
 else
-  echo ${0:t}
+  echo "running cice.batch.csh"
```
[the remark this answers is not in the document]
Comment (phil-blain) [reply configuration/scripts/cice.batch.csh:6]: I explained in the commit message: e6b3f6f

[the remark this answers is not in the document]
Comment (phil-blain) [reply configuration/scripts/cice.batch.csh:6]: I can always change it to `echo "running ${0:t}"`

**Comment by phil-blain** [comment]
OK. I'll update the doc. thanks.

File: configuration/scripts/cice.batch.csh (hunk 1 above)
[the remark this answers is not in the document]
Comment (apcraig) [reply configuration/scripts/cice.batch.csh:6]: I see now and find it curious.  I wonder why we're not using $0 in all the scripts.  I might look into this at some point, but for now, I agree this change is a good idea to make it consistent.  Thanks.

**Comment by phil-blain** [comment]
@apcraig I've updated the documentation. I'll uniformize the wording with CICE-Consortium/Icepack#319 once we converge there.

**Comment by apcraig** [comment]
Looks good.  I have updated the Icepack CPPDEFS documentation and think they're largely consistent.  If you agree, I'll approve the PR and merge it.

**Comment by phil-blain** [comment]
@apcraig I've also added the "using the syntax ``-DCICE_MACRO``" bit here. This should be ready to merge.

**Comment by apcraig** [comment]
Great, I'll merge as soon as the checks are complete.  Looks like readthedocs is struggling today, we may need to retrigger the build in a while.  I am having the same problems with readthedocs with another PR.

**Pull request merged**